packages/web-platform/web-rsbuild-plugin/src/plugins/prefetch-worker.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+
+import type { HtmlRspackPlugin } from '@rsbuild/core';
+import path from 'path';
+import type { Compiler, Compilation } from '@rspack/core';
+
+export class PrefetchWorkerPlugin {
+  resourceHints: HtmlRspackPlugin.HtmlTagObject[] = [];
+  chunkIds: Set<string> = new Set();
+
+  HtmlPlugin: HtmlRspackPlugin;
+
+  constructor(
+    options: { HtmlPlugin: HtmlRspackPlugin },
+  ) {
+    this.HtmlPlugin = options.HtmlPlugin;
+  }
+
+  apply(compiler: Compiler) {
+    if (
+      compiler.options.mode !== 'production'
+      || typeof compiler.options.output?.publicPath === 'function'
+    ) {
+      console.warn(
+        'PrefetchWorkerPlugin only works in production mode and when publicPath is a string',
+      );
+      return;
+    }
+
+    compiler.hooks.compilation.tap(
+      'PrefetchWorkerPlugin',
+      (compilation: Compilation) => {
+        if (!this.HtmlPlugin) {
+          console.warn('HtmlWebpackPlugin not found');
+          return;
+        }
+
+        const htmlPluginHooks = (this.HtmlPlugin as any).getCompilationHooks(
+          compilation,
+        );
+        htmlPluginHooks.alterAssetTags.tap(
+          'PrefetchWorkerPlugin',
+          (data: any) => {
+            this.chunkIds.values().forEach(id => {
+              // [name] is replaced with [id] in chunkFilename(detail: https://rspack.dev/config/output#outputchunkfilename)
+              const outputRelativePath = compilation.getPath(
+                (compilation.options.output.chunkFilename as string)!.replace(
+                  /\[name\]/g,
+                  '[id]',
+                ),
+                {
+                  chunk: Array.from(compilation.chunks).find(i => i.id === id),
+                },
+              );
+              const finalPath = path.join(
```

Comment:
I think this would not work on Windows. Should we use `path.posix` instead?
```suggestion
              const finalPath = path.posix.join(
```